server/src/main/java/org/elasticsearch/gateway/PersistedClusterStateService.java

```diff
@@ -858,7 +852,7 @@ void prepareCommit(
             final Map<String, String> commitData = Maps.newMapWithExpectedSize(COMMIT_DATA_SIZE);
             commitData.put(CURRENT_TERM_KEY, Long.toString(currentTerm));
             commitData.put(LAST_ACCEPTED_VERSION_KEY, Long.toString(lastAcceptedVersion));
-            commitData.put(NODE_VERSION_KEY, Integer.toString(Version.CURRENT.id));
+            commitData.put(NODE_VERSION_KEY, Integer.toString(BuildVersion.current().id()));
```

Comment:
This changes the node version that is stored in metadata in serverless - this doesn't matter, as the BuildVersion implementation there is a no-op